### PR TITLE
registerTextureStream before getTextureStream to avoid frame lost

### PR DESCRIPTION
**Pull request by WengHe**
## Description

Call `registerTextureStream` before getting an input texture stream, with this change we can ensure that when we get an input texture stream, the output stream can be returned to the client immediately, otherwise, the first few frames may be lost if the output stream is not registered yet.

### CODE REVIEW

File: packages/teams-js/src/internal/videoUtils.ts [hunk 1]
```diff
@@ -124,28 +130,38 @@ async function getInputVideoTrack(
...
+ * The function to create a MediaStreamTrack generator.
+ * The generator can then get the processed frames as media stream source.
+ * The generator can be registered back to the media stream so that the host can get the processed frames.
+ */
+function createProcessedStreamGeneratorWithoutSource(): MediaStreamTrack {
+  if (inServerSideRenderingEnvironment()) {
+    throw errorNotSupportedOnPlatform;
+  }
+  const MediaStreamTrackGenerator = window['MediaStreamTrackGenerator'];
+  const generator = new MediaStreamTrackGenerator({ kind: 'video' });
```
Comment (lubobill1990) [inline packages/teams-js/src/internal/videoUtils.ts:143]: return directly

File: change/@microsoft-teams-js-b3109727-ff31-4dcb-bafc-b21748d8ce7b.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Changed the other of registering texture stream",
```
Comment (lubobill1990) [inline change/@microsoft-teams-js-b3109727-ff31-4dcb-bafc-b21748d8ce7b.json:3]: The comment is too vague.

**Review by lubobill1990 — APPROVED** [review]

**Comment by TrevorJoelHarris** [comment]
It seems like there should be some tests that verify that this change is working correctly

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

**Review by AE-MS — CHANGES_REQUESTED** [review]
Just waiting for other feedback to be addressed
:clock1:

**Comment by WengHe** [comment]
> It seems like there should be some tests that verify that this change is working correctly

The change is internal, all the UTs are passed, and I did an end to end test by using the new package in a video app and verified it works well.

File: packages/teams-js/src/internal/videoUtils.ts [hunk 3]
```diff
@@ -124,28 +130,38 @@ async function getInputVideoTrack(
...
-  }
+  generator: MediaStreamTrack,
+): void {
   const MediaStreamTrackProcessor = window['MediaStreamTrackProcessor'];
   const processor = new MediaStreamTrackProcessor({ track: videoTrack });
   const source = processor.readable;
-  const MediaStreamTrackGenerator = window['MediaStreamTrackGenerator'];
-  const generator = new MediaStreamTrackGenerator({ kind: 'video' });
-  const sink = generator.writable;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
[the remark this answers is not in the document]
Comment (WengHe) [reply packages/teams-js/src/internal/videoUtils.ts]: updated, now we don't need it any more

File: change/@microsoft-teams-js-b3109727-ff31-4dcb-bafc-b21748d8ce7b.json (hunk 2 above)
Comment (WengHe) [reply change/@microsoft-teams-js-b3109727-ff31-4dcb-bafc-b21748d8ce7b.json]: updated

File: packages/teams-js/src/internal/videoUtils.ts (hunk 1 above)
Comment (WengHe) [reply packages/teams-js/src/internal/videoUtils.ts]: updated

**Review by TrevorJoelHarris — APPROVED** [review]

**Review by AE-MS — APPROVED** [review]

**Pull request merged**